sympy/series/sequences.py

```diff
@@ -0,0 +1,1144 @@
+from __future__ import print_function, division
+
+from sympy.core.expr import Expr
+from sympy.core.singleton import (S, Singleton)
+from sympy.core.symbol import Dummy
+from sympy.core.function import Lambda
+from sympy.core.add import Add
+from sympy.core.mul import Mul
+from sympy.core.compatibility import (range, integer_types, with_metaclass,\
+                                      is_sequence, iterable, ordered)
+from sympy.core.cache import cacheit
+from sympy.core.sympify import sympify
+from sympy.core.containers import Tuple
+from sympy.core.evaluate import global_evaluate
+from sympy.functions.elementary.integers import ceiling
+from sympy.functions.elementary.miscellaneous import Min, Max
+from sympy.polys import lcm
+from sympy.sets.sets import Interval, Set, Intersection
+from sympy.utilities.iterables import flatten
+
+
+################################################################################
+#                            SEQUENCES                                         #
+################################################################################
+
+
+class SeqBase(Expr):
+    """Base class for sequences"""
+
+    is_iterable = True
+
+    is_EmptySequence = False
+    is_SeqPer = False
+    is_SeqFunc = False
+    is_SeqFormula = False
+    is_SeqAdd = False
+    is_SeqMul = False
```

Comment:
If the same information can be obtaining through `isinstance` checks, then yes they are redundant.
I know that this pattern is used abundantly, but that's not necessarily best practices; in fact it's considered an antipattern in the OO literature, and it got us into trouble already. (That doesn't mean that antipatterns are unavoidable; sometimes they're just the lesser of two evils. However, in the case of SymPy's `is_*` attributes, one of the deeper reasons we're having trouble with them is that we don't know which of them were unavoidable and which of them could be replaced with proper OO functions.)

Grand sweeping considerations aside, you need to document the semantics of each `is_*` property.
The documentation needs to be complete enough so that any Python programmer who sees a use of such an `is_*` property can instantly validate it by looking at the `is_*` property's docstring, *without looking at more than a handful of lines around the use of the `is_*` property*.
You'll need look at each use of any `is_*` property. Also, we'll have to review each such use and validate that it matches the docstring (not looking forward to that; essentially, such properties are convenient for coding but they turn into the opposite for review and maintenance).